src/cart/cart.repository.mongo.ts

```diff
@@ -0,0 +1,58 @@
+import { randomUUID } from "node:crypto";
+import MongoCart, { ICart, ICartItemEntity } from "../schemas/ICart";
+import { Cart, CartBase, CartItem, Product } from "../types";
+import { CartNotFoundError } from "../exceptions/CartNotFoundError";
+import { DatabaseError } from "../exceptions/DatabaseError";
+
+export class CartRepository implements CartBase {
+  async getCart(userId: string): Promise<Cart | ICart> {
+    try {
+      const cart = await MongoCart.findOne({ userId, isDeleted: false }).exec();
+      if (cart) {
+        return cart;
+      }
+      throw new CartNotFoundError();
+    } catch (err: any) {
+      if (err.name === "CartNotFound") {
+        throw err;
+      }
+      throw new DatabaseError(`There was an error fetching cart for user with id ${userId}`);
+    }
+  }
+
+  listCarts(): Cart[] | Promise<Cart[] | ICart[]> {
+    throw new Error("Method not implemented.");
+  }
+  
+  async createCart(userId: string): Promise<Cart | ICart> {
+    try {
+      const newCart = new MongoCart({
+        id: randomUUID(),
+        userId: userId,
+        isDeleted: false,
+        items: []
+      });
+      const savedCart = await newCart.save();
+      return savedCart;
```

Comment:
W: You can just return the value without the await since after the save there are not more code
```suggestion
      return newCart.save();
```